.github/CONTRIBUTING.md

```diff
@@ -1,9 +1,73 @@
 # Contributing to Trino
 
-## Contributor License Agreement ("CLA")
 
-In order to accept your pull request, we need you to [submit a CLA](https://github.com/trinodb/cla).
+This is the process we suggest for contributions.  This process is designed to
+reduce the burden on project reviewers, impact on other contributors, and to
+keep the amount of rework from the contributor to a minimum.
 
-## License
+By contributing to Trino, you agree that your contributions will be licensed
+under the [Apache License Version 2.0 (APLv2)](../LICENSE).
 
-By contributing to Trino, you agree that your contributions will be licensed under the [Apache License Version 2.0 (APLv2)](../LICENSE).
+Find more information at [the Trino site](https://trino.io/development/).
+
+## Contributor License Agreement
+
+In order to accept your pull request, we need you to [submit a contributor
+license agreement (CLA)](https://github.com/trinodb/cla).
+
+## Discussion
+
+Start a discussion by creating a Github
+[issue](https://github.com/trinodb/trino/issues), or asking on
+[Slack](https://trino.io/slack.html) (unless the change is trivial).
+
+* This step helps you identify possible collaborators and reviewers.
+* Does the change align with technical vision and project values?
+* Will the change conflict with another change in progress? If so, work with
+  others to minimize impact.
+* Is this change large?  If so, work with others to break into smaller steps.
+
+## Implementation
+
+Implement the change.
+
+* If the change is large, post a preview Github
+  [pull request](https://github.com/trinodb/trino/pulls) with the title
+  prefixed with `[WIP]`, and share with collaborators.
```

Comment:
why not use Draft instead?

Reply:
I can rephrase that, but we'll have to also update this guide in https://github.com/trinodb/trino.io/blob/master/development/process.md so I'll wait for more feedback to open a PR in https://github.com/trinodb/trino.io to keep both copies in sync.